src/services/service.ts

```diff
@@ -287,6 +288,74 @@ export default class Service {
         return parsed_results
     }
 
+    protected serviceStream<T>(options: ReportOptions, pre_report_hook: PreReportHook) {
+        const query = this.buildCustomerReportQuery(options)
+
+        const call = this.streamSearchData(query)
+
+        return this.streamGenerator<T>(call)
+    }
+
+    private async *streamGenerator<T>(call: ClientReadableStream<SearchGoogleAdsStreamResponse>) {
```

Comment:
what does the `*` do?

Reply:
It means that this function will return a generator.